### PR TITLE
APIGOV-23866 - fix race condition

### PR DESCRIPTION
**Pull request by sbolosan**

### CODE REVIEW

**Comment by sbolosan** [comment]
DRAFT status right now
Trying to figure out race warning issue

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -127,10 +129,13 @@ func InitializeWithAgentFeatures(centralCfg config.CentralConfig, agentFeaturesC
 	if centralCfg.GetUsageReportingConfig().IsOfflineMode() {
 		// Offline mode does not need more initialization
 		agent.cfg = centralCfg
+		setCentralConfig(centralCfg)
 		return nil
 	}
 
 	agent.cfg = centralCfg
```
Comment (vivekschauhan) [inline pkg/agent/agent.go:136]: This can be removed as setCentralConfig sets the variable

File: pkg/agent/agent.go [hunk 2]
```diff
@@ -127,10 +129,13 @@ func InitializeWithAgentFeatures(centralCfg config.CentralConfig, agentFeaturesC
 	if centralCfg.GetUsageReportingConfig().IsOfflineMode() {
 		// Offline mode does not need more initialization
 		agent.cfg = centralCfg
```
Comment (vivekschauhan) [inline pkg/agent/agent.go:131]: This can be removed as setCentralConfig sets the variable

Comment (sbolosan) [reply pkg/agent/agent.go]: thank you

**Review by vivekschauhan — APPROVED** [review]

**Pull request merged**